### ISSUE TITLE
Move btcrrpcclient to btcd repo.

### ISSUE THREAD

**Issue by jrick** [btcsuite/btcd#953]
There is code in the btcd repo (most notably btcctl, but also some of the integration tests) that depends on btcrpcclient.  It would streamline development a lot by moving that package to the btcd repo, as it would be easier to modify it for API changes.

**Comment by jcvernaleo** [comment]
+1 from me.  I have way too many repos to worry about in decred so one less is a good thing.

The only thing I ask is that after it is imported you make a commit to this repo that deletes everything and just leaves a README.md to explain (like I did for decred/ccminer, decred/dcrticketbuyer, and some more of the damned).

**Comment by stevenroose** [comment]
A downside is that all users of btcrpcclient will have to download the whole btcd project as a dependency. It is like Ethereum does it and even though go-ethereum is a lot heavier than btcd, it's a pain in the ass building Docker images that just want to use the tiny HTTP RPC utility.

**Comment by jrick** [comment]
They're pulling it in anyways as a dependency (btcrpcclient depends on btcjson which is in btcd).

**Comment by stevenroose** [comment]
@jrick oh, that's a good point :D And having btcjson makes sense. Hmm, yeah no objection in that case..

**Comment by jrick** [comment]
btcjson isnt' the only one etiher:

```
PS C:\Users\jrick\go\src\github.com\btcsuite\btcrpcclient> (go list -json | convertfrom-json).imports | sls '/btcd/'

github.com/btcsuite/btcrpcclient/vendor/github.com/btcsuite/btcd/btcjson
github.com/btcsuite/btcrpcclient/vendor/github.com/btcsuite/btcd/chaincfg
github.com/btcsuite/btcrpcclient/vendor/github.com/btcsuite/btcd/chaincfg/chainhash
github.com/btcsuite/btcrpcclient/vendor/github.com/btcsuite/btcd/wire
```

**Comment by davecgh** [comment]
In theory, it would be nice if it could be completely separate (no deps) specifically for the reason @stevenroose said, however, it relies on packages already in the btcd repo anyways, and there isn't a good way around that as I'll explain below, so it won't make any difference in that regard.

One of the nice things that `btcrpcclient` provides over something like `gRPC` is that it handles all of the conversions from whatever the underlying types are used to send the data across the wire (JSON-RPC in this case, protobuf in the case of gRPC) into types that already have all of the richness the btcsuite libraries.  This necessarily implies though that it needs access to those libraries and types, so it is sort of a double edged sword in that regard.

Since I'm sure it will get mentioned due to my comments above, it's certainly true that creating a conversion package for `gRPC` (protobuf types -> rich btcsuite types) would be possible too, however, there currently is no such package, nor does btcd itself support `gRPC` yet either anyways.  The result is that every single client has to do all of the conversions manually, and, from my own experience, it is incredibly cumbersome when you have multiple clients all ingesting blocks and transactions all having to handle the conversions themselves.  Anyways, back to `btcrpcclient`.

Perhaps there is a case to be made for separating the client logic from the conversion logic such that the main client would not require a dependency on the btcd repo and, if the caller wanted to use the rich types provided by the btcsuite packages, they could import the conversion logic separately.  That said, after personally having written more than 10 clients as this point, I have not yet run into a situation I don't want access to the richer types.  Therefore, every one of those cases would end up importing from the btcd repository anyways, so splitting the logic would make no difference in the end anyways.

In short, I agree that moving this into the btcd repository is a reasonable thing to do.
